templates/terraform/decoders/backend_signed_url_key.go.erb

```diff
@@ -0,0 +1,27 @@
+keyName := d.Get("name").(string)
+
+policyRaw, ok := res["cdnPolicy"]
```

Comment:
I think this will be a common pattern across fine-grained resources; what do you think of trying to break this into a shared template?

Reply:
mmm I'm not sure how I'd want to do it. Since signed_url_key has the extra step of having a sensitive field that isn't returned, it's probably going to differ a little anyways, and I don't think I've played around enough with the other fine-grained resources we've proposed to feel confident that my changes would cover them. Do you have a good example? 

I guess based on my attempts for two seconds that I don't think it's an easy thing to do, unless we restrict it as much of self_link_query and get the same issue where it doesnt fit what I need (because it only does a list of objects).

Reply:
So far what I have is something like `query_key`, `property_type`, etc which is `cdnPolicy.signedUrlKeyNames`
`peerings`

which we can decide we only handle where `query_key.split(.)[0...-1]` are NestedObjects and [-1] is an Array with type `property_type`, and use that to compare to the `:identity`. If property_type is a string, it just takes the first field from identity? Or uses id? 

OR we take in a full property or ResourceRef and do that - mostly if we want to do this second thing, that's work than I dont want to do in this PR.

Reply:
> Since signed_url_key has the extra step of having a sensitive field that isn't returned, it's probably going to differ a little anyways

Ehh, probably just means that `:ignore_read` will work well on the first revision.

> I don't think I've played around enough with the other fine-grained resources we've proposed to feel confident that my changes would cover them.

No reason not to try now and iterate imo! But if you feel strongly that now isn't the right time, I'll defer to you on that. My feeling on the matter is that we should handle them mostly the "right way" in MM from the start instead of needing to backfill legacy ones afterwards, especially if we need to maintain backwards compatibility across all our providers.

@ the second comment: not following 100% on the later bits, but your approach mirrors what I'd imagine it would look like!